enhancements/machine-api/aws-placement-groups.md

```diff
@@ -0,0 +1,543 @@
+  // zone. This ensures instances each have their own networking and power source
+  // for maximum hardware fault tolerance.
+  // Spread placement groups are recommended for a small number of critical instances
+  // which must be kept separate from one another.
+  // Using a Spread placement group imposes a limit of seven instances within
+  // the placement group within a single availability zone.
+  AWSSpreadPlacementGroupType    AWSPlacementGroupType = "Spread"
+)
+
+// AWSPlacementGroupUnion is a discriminated union of placement group configuration.
+// +union
+type AWSPlacementGroupUnion struct {
+  // GroupType specifies the type of AWS placement group to use for this Machine.
+  // This parameter is only used when a Machine is being created and the named
+  // placement group does not exist.
+  // Valid values are "Cluster", "Partition", "Spread".
+  // If no value is given and a placement group is required to be created, this is
+  // considered an error.
+  // Note: If the value of this field is "Spread", instanceTenancy may not be set
+  // to "dedicated".
+  // +kubebuilder:validation:Enum:="Cluster";"Partition";"Spread"
+  // +unionDiscriminator
+  // +kubebuilder:validation:Required
+  // +required
+  GroupType AWSPlacementGroupType `json:"groupType,omitempty"`
+
+  // +optional
+  Cluster *AWSClusterPlacement `json:"cluster,omitempty"`
+  // +optional
+  Partition *AWSPartitionPlacement `json:"partition,omitempty"`
+  // +optional
+  Spread *AWSSpreadPlacement `json:"spread,omitempty"`
+}
+
+// AWSClusterPlacement defines the configuration for cluster placement groups.
+type AWSClusterPlacement struct {}
+
+// AWSPartitionPlacement defines the configuration for partition placement groups.
+type AWSPartitionPlacement struct {
+  // Count specifies the number of partitions for a Partition placement
+  // group. This value is only observed when creating a placement group and
+  // only when the `groupType` is set to `Partition`.
+  // Note the partition count of a placement group cannot be changed after creation.
+  // If unset, AWS will provide a default partition count.
+  // This default is currently 2.
+  // Note: If instanceTenancy is set to "dedicated", the maximum number of partitions
+  // is limited to 2.
+  // +kubebuilder:validation:Minimum:=1
+  // +kubebuilder:validation:Maximum:=7
+  // +optional
+  Count int32 `json:"count,omitempty"`
+
+  // Number specifies the numbered partition in which instances should be launched.
+  // This value is only used in conjunction with a `Partition` placement Group.
+  // It is recommended to only use this value if multiple MachineSets share
+  // a single Placement Group, in which case, each MachineSet should represent an individual partition number.
+  // If unset, when a Partition placement group is used, AWS will attempt to
+  // distribute instances evenly between partitions.
+  // +kubebuilder:validation:Minimum:=1
```

Comment:
Would it be better to make these pointers so that it can be explicitly omitted? 

Also, we don't actually have API validation on these structs as they aren't CRDs, they are just structs within a `RawExtension`, so I'm not sure how accurate this needs to be